frontend/src/components/WorkoutMenu/WorkoutEditor.tsx

```diff
@@ -30,9 +30,29 @@ import {
 import { useActiveWorkout } from '../../context/ActiveWorkoutContext';
 import { createZoneTableInfo } from '../../utils/zones';
 import { secondsToHoursMinutesAndSecondsString } from '../../utils/time';
+
+const editableWorkoutIsEqualToLoaded = (
+  editable: EditableWorkout,
+  loaded: WorkoutToEdit
+) => {
+  if (editable.name !== loaded.name) return false;
+  if (editable.parts.length !== loaded.parts.length) return false;
+
+  for (let i = 0; i < editable.parts.length; i++) {
+    const editablePart = editable.parts[i];
+    const loadedPart = loaded.parts[i];
+
+    if (editablePart.duration !== loadedPart.duration) return false;
+    if (editablePart.targetPower !== loadedPart.targetPower) return false;
+  }
+
+  return true;
+};
+
```

Comment:
`const editableWorkoutIsEqualToLoaded = (
  editable: EditableWorkout,
  loaded: WorkoutToEdit
) => {
  const getRelevantInfo = (workout: Workout) =>
    ({ name: workout.name, ps: workout.parts.map((part) => [part.duration, part.targetPower]) })

  return getRelevantInfo(editable) === getRelevantInfo(loaded)
};`

I think this should be equivalent. 
You can debate if it is more readable (i think kinda similar), but it is shorter at least 🙃

NVM javascript is a broken language, so this will not work :))
you can ignore :)))!")#!"#)!")#

Reply:
:^) Yeah, I went for the naive/simple way instead of trying to fight JS and potentially making it less readable - even if this result may be clunky